src/test/java/co/cask/cdc/plugins/integration/source/CTSQLServerPluginTest.java

```diff
@@ -0,0 +1,410 @@
+        .put(CTSQLServerConfig.DATABASE_NAME, "non_existing_db")
+        .put(Constants.Reference.REFERENCE_NAME, "CTSQLServerSource")
+        .build();
+      ETLPlugin sourceConfig = new ETLPlugin(PLUGIN_NAME, StreamingSource.PLUGIN_TYPE, sourceProps);
+
+      ETLPlugin sinkConfig = MockSink.getPlugin(outputTable);
+
+      Assertions.assertThatThrownBy(() -> deployETL(sourceConfig, sinkConfig, "testChangesSource"))
+        .isInstanceOf(IllegalStateException.class);
+    }
+
+    @Test
+    public void testDeploymentFailedWithoutTrackingEnabled() throws Exception {
+      try (Connection connection = getConnectionToRoot();
+           Statement statement = connection.createStatement()) {
+        statement.executeUpdate(String.format("create database %s", dbName));
+      }
+
+      ImmutableMap<String, String> sourceProps = ImmutableMap.<String, String>builder()
+        .put(CTSQLServerConfig.HOST_NAME, HOST)
+        .put(CTSQLServerConfig.PORT, PORT)
+        .put(CTSQLServerConfig.USERNAME, USERNAME)
+        .put(CTSQLServerConfig.PASSWORD, PASSWORD)
+        .put(CTSQLServerConfig.DATABASE_NAME, dbName)
+        .put(Constants.Reference.REFERENCE_NAME, "CTSQLServerSource")
+        .build();
+      ETLPlugin sourceConfig = new ETLPlugin(PLUGIN_NAME, StreamingSource.PLUGIN_TYPE, sourceProps);
+
+      ETLPlugin sinkConfig = MockSink.getPlugin(outputTable);
+
+      Assertions.assertThatThrownBy(() -> deployETL(sourceConfig, sinkConfig, "testChangesSource"))
+        .isInstanceOf(IllegalStateException.class);
+    }
+
+  }
+
+  public static class SuccessFlow extends CDCPluginTestBase {
+    @Rule
+    public TestName testName = new TestName();
+
+    private String dbName;
+    private String dbTableName;
+    private String outputTable;
+    private SparkManager programManager;
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+      Class.forName(SQLServerDriver.class.getName());
+    }
+
+    @Before
+    public void setUp() throws Exception {
+      dbName = "CTSQLServerPluginTest_" + testName.getMethodName();
+      dropDatabaseIfExists(dbName);
+      createDatabaseWithTracking(dbName);
+      dbTableName = testName.getMethodName();
+      outputTable = testName.getMethodName() + "_out";
+
+      ImmutableMap<String, String> sourceProps = ImmutableMap.<String, String>builder()
+        .put(CTSQLServerConfig.HOST_NAME, HOST)
+        .put(CTSQLServerConfig.PORT, PORT)
+        .put(CTSQLServerConfig.USERNAME, USERNAME)
+        .put(CTSQLServerConfig.PASSWORD, PASSWORD)
+        .put(CTSQLServerConfig.DATABASE_NAME, dbName)
+        .put(Constants.Reference.REFERENCE_NAME, "CTSQLServerSource")
+        .build();
+      ETLPlugin sourceConfig = new ETLPlugin(PLUGIN_NAME, StreamingSource.PLUGIN_TYPE, sourceProps);
+
+      ETLPlugin sinkConfig = MockSink.getPlugin(outputTable);
+
+      programManager = deployETL(sourceConfig, sinkConfig, "testChangesSource");
+      programManager.start();
```

Comment:
though there is nothing wrong here, use startAndWaitForRun() because it is generally less easy to mis-use than waitForRun(). We might remove waitForRun() at some point in the future.